src/mscorlib/src/System/String.Comparison.cs

```diff
@@ -1144,5 +1144,16 @@ public override int GetHashCode()
 
             return referenceCulture.CompareInfo.IsPrefix(this, value, ignoreCase ? CompareOptions.IgnoreCase : CompareOptions.None);
         }
+
+        [Pure]
+        public Boolean StartsWith(Char value)
```

Comment:
Most other methods on string use `Length`, so I'm guessing the JIT makes it as fast as it can, but it might be worth comparing the code the JIT generates.